db/migrate/20240509190359_add_provisioning_columns_to_model_tables.rb

```diff
@@ -0,0 +1,9 @@
+class AddProvisioningColumnsToModelTables < ActiveRecord::Migration[7.1]
+  def change
+    add_column :model_configs, :provisioned, :boolean, default: false, if_not_exists: true
```

Comment:
Why do you need `if_not_exists` here?

Reply:
You know what, I'd blindly added that as the obvious solution, despite the odd fact that no other migrations needed it. Thinking about it now, it's because I'd initially changed `schema.rb` itself to try things out. In fact, said changed are still in this PR actually; so, I'll remove them.

I'd sort of misinterpreted (_overestimated?_) `db:migrate`. I had this idea that it was implicitly idempotent, when it actually tracks the migrations so it knows never to run them again. While I feel like there's a lane for that to go sideways, and think idempotent migrations are ultimately (_if not, overly_) safer; but, I'll remove the `if_not_exists` as well.

Reply:
`db:migrate` is smart enough to only run migrations that haven't been run on the DB before. You can always assume your migration is being run once.

_Unless_ someone does a `db:rollback`, in which case Rails will try to undo your changes. `add_column` becomes `drop_column`, etc. Then, the next `db:migrate` will run your migration again on that database.